product_docs/docs/pgd/4/harp/09_consensus-layer.mdx

```diff
@@ -142,3 +142,64 @@ instance_defaults:
 The `harp_location` override might also be necessary to favor specific node 
 groupings when using cloud providers such as Amazon that favor availability
 zones in regions over traditional data centers.
+
+## Write Leadership consensus storage
+
+### Write Leader elections
+
+The process of selecting a write leader begins with the startup of the first HARP manager:
+
+* Starts Postgres on the node
+* Reads the Postgres status (including any data on replication lag)
+* Queries the current leader (there should be none)
+* Stores the leadership request in the DCS
+* If the leadership request in not rejected, store the leader lease in the DCS
+* Refresh the leader lease (on a configurable interval, default is 500ms)
+
+Once established, subsequent HARP manager instances startup in a similar fashion:
+
+* Start Postgres on the node
+* Reads the Postgres status (including any data on replication lag)
+* Queries the current leader (there should be one existing)
+* As there is a leader, assume a shadow role and monitor the leader lease
+* If the lease expires, follow the path of the first HARP manager instance to claim leadership
+
+When HARP Proxy instances are started, they configure themselves with this logic:
+
+* Subscribe to and watch for key changes in the DCS
+* If the leader changes, reconfigure routing
+
+The sequence of events is shown in the diagram below:
+
+![Sequence Diagram](images/HARPsequencediagram.png)
+
+### Detection and recovery
+
+When partitions in availability zones (AZ) occur, the process detailed in Leader elections leads to the following behaviors:
+
+* **Partition of an AZ containing the write leader**
+  * Leader lease expires
+  * New leader is elected by the remaining nodes
+  * Synchronous replication is reconfigured
+
+
+* **Partition of a AZ not containing the write leader**
+  * The monitor function detects the loss of synchronous commit consensus (on a 1 second loop)
```

Comment:
Nothing happens here in normal case, this is MC specific only

Reply:
So this should say that things continue to work normally as long as the leader still has majority.